### PR TITLE
Update README for the multi-dc example

### PR DESCRIPTION
**Pull request by ryantwolf**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Updates the README to describe what the multi-datacenter example is and how to use it.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Updates the README to describe what the multi-datacenter example is and how to use it.

# Usage

N/A

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: examples/llm/pretrain/README.md [hunk 1]
```diff
@@ -69,4 +69,14 @@ When running the custom_recipe.py file, it will execute the `custom_llama3_8b` r
    python custom_recipe.py "trainer.max_steps=*2"
    ```
 
-These options provide flexibility in customizing your pretraining recipe directly from the command line.
\ No newline at end of file
+These options provide flexibility in customizing your pretraining recipe directly from the command line.
```
Comment (jgerh) [inline examples/llm/pretrain/README.md:72]: Fix punctuation

These options provide flexibility in customizing your pre-training recipe directly from the command line.

File: examples/llm/pretrain/README.md [hunk 2]
```diff
@@ -69,4 +69,14 @@ When running the custom_recipe.py file, it will execute the `custom_llama3_8b` r
...
+
+## Multi-datacenter Example
+`multi_datacenter.py` provides a modified Llama 3.1 70B recipe with a few modifications to boost its performance across 2 datacenters.
+1. 2 distributed optimizer instances. This is achieved by setting `num_distributed_optimizer_instances=2` in `nemo.lightning.MegatronStrategy`.
+1. `net_name` as `socket` for the `inter_dp_cp` communication group. This is achieved through the NCCL config file `multi_dc_nccl_communicator_config.yaml` and passing its path to `nemo.lightning.MegatronStrategy`.
```
Comment (jgerh) [inline examples/llm/pretrain/README.md:77]: suggested revision for clarity

The `multi_datacenter.py`script modifies the Llama 3.1 70B recipe to boost its performance across two datacenters by:
1.	Enabling two distributed optimizer instances: set `num_distributed_optimizer_instances=2` in `nemo.lightning.MegatronStrategy`.
2.	Configuring inter_dp_cp communication via socket: Use the `multi_dc_nccl_communicator_config.yaml NCCL file (which sets net_name to socket) and pass its path to `nemo.lightning.MegatronStrategy`.

File: examples/llm/pretrain/README.md [hunk 3]
```diff
@@ -69,4 +69,14 @@ When running the custom_recipe.py file, it will execute the `custom_llama3_8b` r
...
+
+Run the example recipe with this command.
```
Comment (jgerh) [inline examples/llm/pretrain/README.md:79]: Fix punctuation

Run the example recipe with this command:

File: examples/llm/pretrain/README.md [hunk 4]
```diff
@@ -69,4 +69,14 @@ When running the custom_recipe.py file, it will execute the `custom_llama3_8b` r
...
+```bash
+python multi_datacenter.py
+```
```
Comment (jgerh) [inline examples/llm/pretrain/README.md:82]: I am providing additional edits and suggested text revisions for the rest of the  file (not just the new content). Edits are identified by line number. 

Line 1/1
I recommend writing a more descriptive title for SEO. “Pre-training” can apply to multiple procedures. Here are some suggestions:

# Pre-train LLMs with Default and Custom Recipes
# Get Started with LLM Pre-training
# Pre-training Recipes

Line 2/2
Add an introductory paragraph that describes the purpose of this doc.  here is the suggested intro:

This guide will walk you through the process of pre-training LLMs using NeMo Framework. We'll cover how to use the NeMo CLI to list available pre-training recipes, run them with default or custom configurations, and even create your own custom recipes using Python scripts. You'll also find examples for inspecting recipes interactively and running training across multiple datacenters.

Line 3/3
Revise title with an imperative verb and bump heading level up from ### to ##

## List Pre-training Recipes

Line 12/12
Fix capitalization in title and bump heading level up from ### to ##

## Run Pre-training with a Default Recipe

Line 35/35
Revise sentence and fix punctuation, capitalization

The syntax of the CLI is the same as the Python code. Which is great, but in some cases, you might want to inspect and edit a recipe interactively. An easy way to do this with the CLI is to use the --repl flag.

Line 43/43
Revise sentences and fix capitalization 

Alternatively, you can trigger a pre-training run from a Jupyter Notebook. See [pretrain.ipynb](pretrain.ipynb) for an example. This method also allows for visualizing all configurations in a structured format, such as the `llama3_8b` recipe shown below:

Line 48/48
Fix capitalization in title and bump heading level up from ### to ##

## Create and Run a Custom Recipe

Line 52/52
Fix punctuation

Note that we end the script with a call to `run.cli.main()`, which uses the same syntax as the CLI, but allows us to provide specific defaults.

Line 67/67
Revise step text for consistency with previous steps

3. To apply transformations when triggering the CLI as if it's Python code:

**Review by jgerh — COMMENTED** [review]
I completed the review of the new text. I am providing additional edits and suggested text revisions for the rest of the file (not just the new content). Edits are identified by line number. Please address all edits in the file.

File: examples/llm/pretrain/README.md [hunk 5]
```diff
@@ -69,4 +69,14 @@ When running the custom_recipe.py file, it will execute the `custom_llama3_8b` r
...
 
-These options provide flexibility in customizing your pretraining recipe directly from the command line.
\ No newline at end of file
+These options provide flexibility in customizing your pretraining recipe directly from the command line.
+
+## Multi-datacenter Example
+`multi_datacenter.py` provides a modified Llama 3.1 70B recipe with a few modifications to boost its performance across 2 datacenters.
+1. 2 distributed optimizer instances. This is achieved by setting `num_distributed_optimizer_instances=2` in `nemo.lightning.MegatronStrategy`.
+1. `net_name` as `socket` for the `inter_dp_cp` communication group. This is achieved through the NCCL config file `multi_dc_nccl_communicator_config.yaml` and passing its path to `nemo.lightning.MegatronStrategy`.
+
```
Comment (xrennvidia) [inline examples/llm/pretrain/README.md:78]: Not sure which MCore version the release will use, but `inter_dp_cp` does not exist in lates MCore any more, it's now [inter_ep_dp](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/parallel_state.py#L1245).

**Review by xrennvidia — COMMENTED** [review]
Your last PR does not handle "use_tp_pp_dp_mapping", better to add it in this PR.
